azurelinuxagent/common/event.py

```diff
@@ -581,24 +580,23 @@ def _trim_extension_event_parameters(event):
         event.parameters = trimmed_params
 
     @staticmethod
-    def report_dropped_events_error(count, errors, op, max_errors_to_report): # pylint: disable=C0103
+    def report_dropped_events_error(count, errors, operation_name, max_errors_to_report):
         err_msg_format = "DroppedEventsCount: {0}\nReasons (first {1} errors): {2}"
         if count > 0:
-            add_event(op=op,
+            add_event(op=operation_name,
                       message=err_msg_format.format(count, max_errors_to_report, ', '.join(errors)),
                       is_success=False)
 
-    def collect_events(self): # pylint: disable=R0914
+    def collect_events(self, enqueue_event_func): # pylint: disable=R0914
         """
         Retuns a list of events that need to be sent to the telemetry pipeline and deletes the corresponding files
         from the events directory.
         """
         max_collect_errors_to_report = 5
-        event_list = TelemetryEventList()
         event_directory_full_path = os.path.join(conf.get_lib_dir(), EVENTS_DIRECTORY)
         event_files = os.listdir(event_directory_full_path)
-        unicode_error_count, unicode_errors = 0, []
-        collect_event_error_count, collect_event_errors = 0, []
+        unicode_error_count, unicode_errors = 0, set()
```

Comment:
Changed the error catching to a set to ensure we catch unique errors rather than the same one

Reply:
I have a few questions about the information we capture when we drop events; let's sync offline